tests/e2e/magic_functions/functions/c_read.cpp

```diff
@@ -0,0 +1,134 @@
+// Copyright 2022 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+#include "mg_procedure.h"
+
+static mgp_func_result *return_function_argument(struct mgp_list *args, mgp_func_context *ctx,
+                                                 struct mgp_memory *memory) {
+  mgp_func_result *result{};
+
+  mgp_value *value{};
+  auto err_code = mgp_list_at(args, 0, &value);
+  if (err_code != MGP_ERROR_NO_ERROR) {
+    mgp_func_result_error("Failed to fetch list!", memory, &result);
+    return result;
+  }
+
+  err_code = mgp_func_result_value(value, memory, &result);
+  if (err_code != MGP_ERROR_NO_ERROR) {
+    mgp_func_result_error("Failed to construct return value!", memory, &result);
+    return result;
+  }
+  return result;
+}
+
+double get_element_from_arg(struct mgp_list *args, int index) {
+  mgp_value *value{};
+  mgp_list_at(args, index, &value);
+
+  double result;
+  int is_int;
+  mgp_value_is_int(value, &is_int);
+
+  if (is_int) {
+    int64_t result_int;
+    mgp_value_get_int(value, &result_int);
+    result = static_cast<double>(result_int);
+  } else {
+    mgp_value_get_double(value, &result);
+  }
+  return result;
+}
+
+static mgp_func_result *add_two_numbers(struct mgp_list *args, mgp_func_context *ctx, struct mgp_memory *memory) {
+  mgp_func_result *result{};
+
+  auto first = get_element_from_arg(args, 0);
+  auto second = get_element_from_arg(args, 1);
+
+  mgp_value *value{};
```

Comment:
```suggestion
  mgp_value *value{nullptr};
```

I think it is not initialized to `nullptr` by default, but I might be wrong.